tool/tsh/aliases.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/google/shlex"
+	"github.com/gravitational/trace"
+)
+
+const tshAliasEnvKey = "TSH_ALIAS"
+
+// tryRunAlias inspects the arguments to see if the alias command should be run, and does so if required.
+func tryRunAlias(ctx context.Context, currentExecPath string, aliases map[string]string, args []string, setEnv func(key, value string) error, getEnv func(key string) string) (bool, string, error) {
```

Comment:
Can you provide some `tsh` tests for the full flow of  `tryRunAlias` function ?

Reply:
I've added test coverage. `tryRunAlias` no longer exists because of refactoring, but the equivalent functionality is now tested.